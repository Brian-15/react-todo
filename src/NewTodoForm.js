import { useState } from 'react';

const NewTodoForm = ({ addTodo }) => {
    const [ textData, setTextData ] = useState('');

    const handleChange = evt => {
        const { value } = evt.target;
        setTextData(value);
    };

    const handleSubmit = evt => {
        evt.preventDefault();
        if (textData === '') return;
        addTodo(textData);
        setTextData('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor='text'>Text
                <input onChange={handleChange} name='text' value={textData} />
            </label>
            <button>Add Task</button>
        </form>
    );
};

export default NewTodoForm;