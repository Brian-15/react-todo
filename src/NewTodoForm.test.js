import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm.js';

describe('Todo component', () => {
    test('smoke test: renders component', () => {
        render(<NewTodoForm />);
    });

    test('snapshot test', () => {
        const { asFragment } = render(<NewTodoForm />);
        expect(asFragment()).toMatchSnapshot();
    });
});