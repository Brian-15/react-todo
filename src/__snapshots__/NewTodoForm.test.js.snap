// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component snapshot test 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Text
      <input
        data-testid="text-input"
        name="text"
        value=""
      />
    </label>
    <button>
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
